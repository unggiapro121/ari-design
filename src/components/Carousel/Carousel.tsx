import { Options, Splide, SplideProps, SplideSlide, SplideTrack } from '@splidejs/react-splide';
import clsx from 'clsx';
import React from 'react';

import iconChevronRight from '@/images/icons/icon-chevron-right.svg';

import Icon from '../Icon/Icon';

import '@splidejs/react-splide/css/core';
import styles from './Carousel.module.scss';

export interface CarouselProps {
  /**
   * Toggle back/forward arrow visibility
   */
  showArrows?: boolean;
  /**
   * Automatically progress through slides. Default interval is `5000` (milliseconds).
   * Use `options` key `interval` to override.
   */
  autoPlay?: boolean;
  /**
   * Splide configuration object - default settings include
   * `autoWidth: true`, `focus: 'center'`, `type: 'loop'`.
   * For all options see documentation https://splidejs.com/guides/options/#options
   */
  options?: SplideProps['options'];
  /**
   * Custom html element tag for the carousel container
   */
  tag?: SplideProps['tag'];
  /**
   * Child elemements are rendered into the carousel slides
   */
  children: SplideProps['children'];
}

/**
 * Unopinionated Carousel wrapper component, supporting fully responsive child elements as slides.
 * Carousel should be automatically sized based on the size of the child items.
 * Plugin: Splide JS - splidejs.com
 */
const Carousel = ({ children, options, showArrows = true, autoPlay = false }: CarouselProps) => {
  const defaultOptions: Options = {
    autoplay: autoPlay,
    autoWidth: true,
    focus: 'center',
    type: 'loop',
    perPage: 1,
    flickMaxPages: 1,
    updateOnMove: true,
  };

  const Slides = () => {
    if (!children) return null;

    return <SplideTrack>{
      React.Children.toArray(children)
        .map((item, key) => (
          <SplideSlide key={key}>
            {item}
          </SplideSlide>
        ))
    }</SplideTrack>;
  };

  const Arrows = () => {
    return (
      <div className={clsx(styles.arrowContainer, !showArrows && styles.arrowContainer__noShow, 'splide__arrows')}>
        <button className={clsx(styles.arrowPrev, 'splide__arrow', 'splide__arrow--prev')}>
          <Icon size={24} url={iconChevronRight} aria-hidden />
        </button>
        <button className={clsx(styles.arrowNext, 'splide__arrow', 'splide__arrow--next')}>
          <Icon size={24} url={iconChevronRight} aria-hidden />
        </button>
      </div>
    );
  };

  return (
    <div data-testid="Carousel">
      <Splide hasTrack={false} className={styles.carousel} options={{
        ...defaultOptions,
        ...options,
      }}>
        <Slides />
        <Arrows />
      </Splide>
    </div>
  );
};

export default Carousel;
