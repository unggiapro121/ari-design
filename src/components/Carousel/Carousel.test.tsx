import { render } from '@testing-library/react';
import React from 'react';

import { CardsWithArrows } from './Carousel.stories';

describe('Carousel component', () => {
  test('renders story', () => {
    const story = render(<CardsWithArrows options={{ type: 'slide' }} />);

    expect(story.getByTestId('Carousel')).toBeTruthy();
    expect(story.getAllByRole('tabpanel').length).toEqual(4);
    expect(story.getAllByRole('heading')[2].textContent).toEqual('Card 3');
  });

  test('snapshot of story', () => {
    const story = render(<CardsWithArrows />);

    expect(story).toMatchSnapshot();
  });

  test('has prev/next buttons', () => {
    const story = render(<CardsWithArrows />);
    const btnNext = story.getByLabelText('Next slide');
    const btnPrev = story.getByLabelText('Go to last slide');

    expect(btnNext).toBeTruthy();
    expect(btnNext?.tagName).toEqual('BUTTON');
    expect(btnPrev).toBeTruthy();
    expect(btnPrev?.tagName).toEqual('BUTTON');
  });

  test('has slide indicators', () => {
    const story = render(<CardsWithArrows />);
    const indicators = story.getByRole('tablist');
    const btns = story.getAllByRole('tab');

    expect(indicators).toBeTruthy();
    expect(btns).toBeTruthy();
    expect(btns[0].tagName).toEqual('BUTTON');
    expect(btns[0].hasAttribute('aria-selected')).toBeTruthy();
  });
});
