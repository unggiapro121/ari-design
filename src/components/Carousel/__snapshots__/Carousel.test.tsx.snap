// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel component > snapshot of story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="Carousel"
      >
        <div
          aria-roledescription="carousel"
          class="splide _carousel_e10190 splide--loop splide--ltr splide--draggable is-active is-initialized"
          id="splide02"
          role="region"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="splide__track splide__track--loop splide__track--ltr splide__track--draggable"
            id="splide02-track"
            style="padding-left: 0px; padding-right: 0px;"
          >
            <ul
              class="splide__list"
              id="splide02-list"
              role="presentation"
              style="transform: translateX(0px);"
            >
              <li
                aria-label="1 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-active is-visible"
                id="splide02-clone01"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 1
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="2 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible"
                id="splide02-clone02"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 2
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="3 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible"
                id="splide02-clone03"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 3
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="4 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible is-prev"
                id="splide02-clone04"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 4
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="1 of 4"
                aria-roledescription="slide"
                class="splide__slide is-active is-visible"
                id="splide02-slide01"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 1
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="2 of 4"
                aria-roledescription="slide"
                class="splide__slide is-visible is-next"
                id="splide02-slide02"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 2
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="3 of 4"
                aria-roledescription="slide"
                class="splide__slide is-visible"
                id="splide02-slide03"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 3
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="4 of 4"
                aria-roledescription="slide"
                class="splide__slide is-visible"
                id="splide02-slide04"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 4
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="1 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-active is-visible"
                id="splide02-clone05"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 1
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="2 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible"
                id="splide02-clone06"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 2
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="3 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible"
                id="splide02-clone07"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 3
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
              <li
                aria-label="4 of 4"
                aria-roledescription="slide"
                class="splide__slide splide__slide--clone is-visible"
                id="splide02-clone08"
                role="tabpanel"
              >
                <div
                  style="width: 640px; max-width: 100vw;"
                >
                  <div
                    class="_card_d75732 undefined"
                    data-testid="Card"
                  >
                    <h1>
                      Card 4
                    </h1>
                    <p>
                      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                    </p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
          <div
            class="_arrowContainer_e10190 splide__arrows splide__arrows--ltr"
          >
            <button
              aria-controls="splide02-track"
              aria-label="Go to last slide"
              class="_arrowPrev_e10190 splide__arrow splide__arrow--prev"
            >
              <img
                alt=""
                aria-hidden="true"
                height="24"
                src="/src/images/icons/icon-chevron-right.svg"
                width="24"
              />
            </button>
            <button
              aria-controls="splide02-track"
              aria-label="Next slide"
              class="_arrowNext_e10190 splide__arrow splide__arrow--next"
            >
              <img
                alt=""
                aria-hidden="true"
                height="24"
                src="/src/images/icons/icon-chevron-right.svg"
                width="24"
              />
            </button>
          </div>
          <ul
            aria-label="Select a slide to show"
            class="splide__pagination splide__pagination--ltr"
            role="tablist"
          >
            <li
              role="presentation"
            >
              <button
                aria-controls="splide02-slide01"
                aria-label="Go to slide 1"
                aria-selected="true"
                class="splide__pagination__page is-active"
                role="tab"
                type="button"
              />
            </li>
            <li
              role="presentation"
            >
              <button
                aria-controls="splide02-slide02"
                aria-label="Go to slide 2"
                class="splide__pagination__page"
                role="tab"
                tabindex="-1"
                type="button"
              />
            </li>
            <li
              role="presentation"
            >
              <button
                aria-controls="splide02-slide03"
                aria-label="Go to slide 3"
                class="splide__pagination__page"
                role="tab"
                tabindex="-1"
                type="button"
              />
            </li>
            <li
              role="presentation"
            >
              <button
                aria-controls="splide02-slide04"
                aria-label="Go to slide 4"
                class="splide__pagination__page"
                role="tab"
                tabindex="-1"
                type="button"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="Carousel"
    >
      <div
        aria-roledescription="carousel"
        class="splide _carousel_e10190 splide--loop splide--ltr splide--draggable is-active is-initialized"
        id="splide02"
        role="region"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          class="splide__track splide__track--loop splide__track--ltr splide__track--draggable"
          id="splide02-track"
          style="padding-left: 0px; padding-right: 0px;"
        >
          <ul
            class="splide__list"
            id="splide02-list"
            role="presentation"
            style="transform: translateX(0px);"
          >
            <li
              aria-label="1 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-active is-visible"
              id="splide02-clone01"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 1
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="2 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible"
              id="splide02-clone02"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 2
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="3 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible"
              id="splide02-clone03"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 3
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="4 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible is-prev"
              id="splide02-clone04"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 4
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="1 of 4"
              aria-roledescription="slide"
              class="splide__slide is-active is-visible"
              id="splide02-slide01"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 1
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="2 of 4"
              aria-roledescription="slide"
              class="splide__slide is-visible is-next"
              id="splide02-slide02"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 2
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="3 of 4"
              aria-roledescription="slide"
              class="splide__slide is-visible"
              id="splide02-slide03"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 3
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="4 of 4"
              aria-roledescription="slide"
              class="splide__slide is-visible"
              id="splide02-slide04"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 4
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="1 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-active is-visible"
              id="splide02-clone05"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 1
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="2 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible"
              id="splide02-clone06"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 2
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="3 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible"
              id="splide02-clone07"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 3
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
            <li
              aria-label="4 of 4"
              aria-roledescription="slide"
              class="splide__slide splide__slide--clone is-visible"
              id="splide02-clone08"
              role="tabpanel"
            >
              <div
                style="width: 640px; max-width: 100vw;"
              >
                <div
                  class="_card_d75732 undefined"
                  data-testid="Card"
                >
                  <h1>
                    Card 4
                  </h1>
                  <p>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, 
                  </p>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="_arrowContainer_e10190 splide__arrows splide__arrows--ltr"
        >
          <button
            aria-controls="splide02-track"
            aria-label="Go to last slide"
            class="_arrowPrev_e10190 splide__arrow splide__arrow--prev"
          >
            <img
              alt=""
              aria-hidden="true"
              height="24"
              src="/src/images/icons/icon-chevron-right.svg"
              width="24"
            />
          </button>
          <button
            aria-controls="splide02-track"
            aria-label="Next slide"
            class="_arrowNext_e10190 splide__arrow splide__arrow--next"
          >
            <img
              alt=""
              aria-hidden="true"
              height="24"
              src="/src/images/icons/icon-chevron-right.svg"
              width="24"
            />
          </button>
        </div>
        <ul
          aria-label="Select a slide to show"
          class="splide__pagination splide__pagination--ltr"
          role="tablist"
        >
          <li
            role="presentation"
          >
            <button
              aria-controls="splide02-slide01"
              aria-label="Go to slide 1"
              aria-selected="true"
              class="splide__pagination__page is-active"
              role="tab"
              type="button"
            />
          </li>
          <li
            role="presentation"
          >
            <button
              aria-controls="splide02-slide02"
              aria-label="Go to slide 2"
              class="splide__pagination__page"
              role="tab"
              tabindex="-1"
              type="button"
            />
          </li>
          <li
            role="presentation"
          >
            <button
              aria-controls="splide02-slide03"
              aria-label="Go to slide 3"
              class="splide__pagination__page"
              role="tab"
              tabindex="-1"
              type="button"
            />
          </li>
          <li
            role="presentation"
          >
            <button
              aria-controls="splide02-slide04"
              aria-label="Go to slide 4"
              class="splide__pagination__page"
              role="tab"
              tabindex="-1"
              type="button"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
